targets/tofino/test/programs/research/CMakeLists.txt

```diff
@@ -139,22 +139,21 @@ p4tools_add_tests(
 p4tools_add_xfail_reason(
   "flay-tofino1-research-tna"
   ".*"
-  switcharoo.p4
+  switcharoo.p4 # Could not infer a type for parameter index (inferred type is don't care '_')
   # scion.p4
   # aes_1pipe.p4
   # cmac_1pipe.p4
-  RTT.p4
-  PRECISION.p4
-  p40f_tofino.p4
+  RTT.p4 # Could not infer a type for parameter index (inferred type is don't care '_')
```

Comment:
This type of error is due to the unspecified index type, e.g. `Register<bit<32>,_>(32w1) reg_ins_ctr`. Can we infer the type from the size `32w1`? (I cannot find related specification from Tofino's doc)

Reply:
Maybe we need to change the source programs. It could be because of a compiler change. If that is the case we should maybe copy the P4 programs into the research folder.

Reply:
Maybe, we just need to modify the Tofino include, that could help..

Reply:
Changing the source programs sounds fine, but we may need to create patches for ease of testing. 

> Maybe, we just need to modify the Tofino include, that could help..
Hmm, how does modifying the Tofino include solve this?

Reply:
I actually ran into this before, thankfully I created an issue: https://github.com/p4lang/p4c/issues/3934

The only way to fix it is to modify the program.

Reply:
So as you suggested patch sets are the best idea. But it will be a little tedious.

Reply:
I have created an example patch set for one program.

Reply:
Great! I can add patches for the rest of them.

Reply:
Thanks!